src/shared/components/oncoprint/TooltipUtils.ts

```diff
@@ -123,34 +124,72 @@ export function makeClinicalTrackTooltip(track:ClinicalTrackSpec, link_id?:boole
 }
 export function makeHeatmapTrackTooltip(genetic_alteration_type:MolecularProfile["molecularAlterationType"], link_id?:boolean) {
     return function (dataUnderMouse:any[]) {
+        
         let data_header = '';
-        let profile_data = 'N/A';
+        let valueTextElement = 'N/A';
+        let categoryTextElement = '';
+
         switch(genetic_alteration_type) {
-            case "MRNA_EXPRESSION":
+            case AlterationTypeConstants.MRNA_EXPRESSION:
                 data_header = 'MRNA: ';
                 break;
-            case "PROTEIN_LEVEL":
+            case AlterationTypeConstants.PROTEIN_LEVEL:
                 data_header = 'PROT: ';
                 break;
-            case "METHYLATION":
+            case AlterationTypeConstants.METHYLATION:
                 data_header = 'METHYLATION: ';
                 break;
+            case AlterationTypeConstants.TREATMENT_RESPONSE:
+                data_header = 'TREATMENT: ';
+                break;
         }
+
         let profileDataSum = 0;
+        const profileCategories:string[] = [];
         let profileDataCount = 0;
+        let categoryCount = 0;
         for (const d of dataUnderMouse) {
             if ((d.profile_data !== null) && (typeof d.profile_data !== "undefined")) {
-                profileDataSum += d.profile_data;
-                profileDataCount += 1;
+                if (genetic_alteration_type === AlterationTypeConstants.TREATMENT_RESPONSE && d.category) {
+                    profileCategories.push(d.category);
+                    categoryCount += 1;
+                } else {
+                    profileDataSum += d.profile_data;
+                    profileDataCount += 1;
+                }
             }
         }
+        
         if (profileDataCount > 0) {
-            profile_data = (profileDataSum/profileDataCount).toFixed(2);
-            if (profileDataCount > 1) {
-                profile_data = `${profile_data} (average of ${profileDataCount} values)`;
+            const profileDisplayValue = (profileDataSum/profileDataCount).toFixed(2);
+            if (profileDataCount === 1) {
+                valueTextElement = profileDisplayValue;
+            } else {
+                valueTextElement = `${profileDisplayValue} (average of ${profileDataCount} values)`;
+            }
+        }
+        
+        if (categoryCount > 0) {
+            if (profileDataCount === 0 && categoryCount === 1) {
+                categoryTextElement = profileCategories[0];
+            } else if (profileDataCount > 0 && categoryCount === 1) {
+                categoryTextElement = `${profileCategories[0]}`;
+            } else {
+                categoryTextElement = `${_.uniq(profileCategories).join(", ")} (${categoryCount} data points)`;
             }
         }
-        let ret = data_header + '<b>' + profile_data + '</b><br/>';
+
+        let ret = data_header;
+        if (valueTextElement !== 'N/A' || categoryCount === 0) {
```

Comment:
it would be better to pull "N/A" out into a constant and use it both here and in the place where you construct your data. It gets messy to deal with checking for string equality everywhere.

Reply:
Ok, changed